src/qpmodel.jl

```diff
@@ -50,9 +50,10 @@ function QuadraticModel(c :: AbstractVector,
 end
 
 function QuadraticModel(c :: AbstractVector, H :: AbstractMatrix;
-                        A :: AbstractMatrix=zeros(0,length(c)), lcon :: AbstractVector=zeros(0), ucon :: AbstractVector=zeros(0),
-                        lvar :: AbstractVector = fill(-Inf, length(c)), uvar :: AbstractVector = fill(Inf, length(c)),
-                        c0 :: Float64=0.0, kwargs...)
+                        A :: AbstractMatrix=zeros(eltype(c), 0,length(c)),
+                        lcon :: AbstractVector=zeros(eltype(c), 0), ucon :: AbstractVector=zeros(eltype(c), 0),
+                        lvar :: AbstractVector = fill(eltype(c)(-Inf), length(c)), uvar :: AbstractVector = fill(eltype(c)(Inf), length(c)),
+                        c0 :: T = zero(eltype(c)), kwargs...) where T
```

Comment:
Same here.

Reply:
Should we use a parametric type then, e.g., `QuadraticModel{T}`? That would mean that we also need `AbstractNLPModels{T}`.

Reply:
It is not required for more abstract types to have parametric types, to have parametric types on the concrete types.
And parametric types on NLPs might break mixed-precision methods. Also, we can argue that `f(x) = x^2` is not tied to a type, but QPData is tied to arrays that have `eltype`s, so it is tied to those types. The question is more on whether we want to accept `eltype(A) = Float32` and `eltype(Q) == Float64` at the time or not.

Reply:
We don't.

Reply:
Then we can use use `T` for every array. :+1: